ReactSkia/components/RSkComponent.cpp

```diff
@@ -106,43 +106,47 @@ RnsShell::LayerInvalidateMask RSkComponent::updateProps(const ShadowView &newSha
    }
   //ShadowOpacity
    if ((forceUpdate) || (oldviewProps.shadowOpacity != newviewProps.shadowOpacity)) {
-      layer_->shadowOpacity = ((newviewProps.shadowOpacity > 1.0) ? 1.0:newviewProps.shadowOpacity)*MAX_8BIT;
+      layer_->componentShadow.shadowOpacity = ((newviewProps.shadowOpacity > 1.0) ? 1.0:newviewProps.shadowOpacity)*MAX_8BIT;
       /*TODO : To be tested and confirm updateMask need for this Prop*/
       updateMask =static_cast<RnsShell::LayerInvalidateMask>(updateMask | RnsShell::LayerInvalidateAll);
       createShadowFilter=true;
    }
   //shadowRadius
    if ((forceUpdate) || (oldviewProps.shadowRadius != newviewProps.shadowRadius)) {
-      layer_->shadowRadius = newviewProps.shadowRadius;
+      layer_->componentShadow.shadowRadius = newviewProps.shadowRadius;
       /*TODO : To be tested and confirm updateMask need for this Prop*/
       updateMask =static_cast<RnsShell::LayerInvalidateMask>(updateMask | RnsShell::LayerInvalidateAll);
       createShadowFilter=true;
    }
   //shadowoffset
    if ((forceUpdate) || (oldviewProps.shadowOffset != newviewProps.shadowOffset)) {
-      layer_->shadowOffset = RSkSkSizeFromSize(newviewProps.shadowOffset);
+      layer_->componentShadow.shadowOffset = RSkSkSizeFromSize(newviewProps.shadowOffset);
       /*TODO : To be tested and confirm updateMask need for this Prop*/
       updateMask =static_cast<RnsShell::LayerInvalidateMask>(updateMask | RnsShell::LayerInvalidateAll);
       createShadowFilter=true;
    }
   //shadowcolor
    if ((forceUpdate) || (oldviewProps.shadowColor != newviewProps.shadowColor)) {
-      layer_->shadowColor = RSkColorFromSharedColor(newviewProps.shadowColor,SK_ColorBLACK);
+      layer_->componentShadow.shadowColor = RSkColorFromSharedColor(newviewProps.shadowColor,SK_ColorBLACK);
       /*TODO : To be tested and confirm updateMask need for this Prop*/
       updateMask =static_cast<RnsShell::LayerInvalidateMask>(updateMask | RnsShell::LayerInvalidateAll);
       createShadowFilter=true;
    }
-
+  //Reset Show Create flasg, if Shadow won't not be visible
+   if(layer_->componentShadow.shadowOffset.isEmpty() && (!layer_->opacity)) {
+      createShadowFilter=true;
+   }
    /* Reset shadow filter - if valid and shadow opacity is 0.0 */
    /* Create shadow filter - shadow opacity is not 0 and change in any props of shadow - opacity,offset,radius */
```

Comment:
Can this comment be updated to reflect the changes .

Resetting shadow filter here on below scenario  and creation/Re-creation happens in components:

- when Shadow won't be visible outside the component., where shadow is directly under the component [Shadow Offset as Zero]
- Shadow is fully transparent
- Shadow will be visible , but it's property have changed.